dali/test/python/test_operator_arithmetic_ops.py

```diff
@@ -123,34 +146,50 @@ def div_promote(left_type, right_type):
     return float_bin_promote(left_dtype, right_dtype).type
 
 
-def int_generator(shape, type):
+def int_generator(shape, type, no_zeros):
     iinfo = np.iinfo(type)
     result = np.random.randint(iinfo.min / 2, iinfo.max / 2, shape, type)
     zero_mask = result == 0
-    return result + zero_mask
-
-
-def float_generator(shape, type):
+    if no_zeros:
+        return result + zero_mask
+    return result
+
+def bool_generator(shape, no_zeros):
+    result = np.random.choice(a=[True, False], size=shape)
+    zero_mask = result == False
+    if no_zeros:
+        return result | zero_mask
+    return result
+
+def float_generator(shape, type, _):
     if (len(shape) == 2):
         return type(np.random.rand(*shape))
     else:
         return type([np.random.rand()])
 
+# Generates inputs of required shapes and types
+# The number of inputs is based on the lenghth of tuple `types`, if types is a single elementm
+# it is considered we should generate 1 output.
+# If the kind contains 'scalar', than the result is batch of scalar tensors.
+# the "shape" of `kinds` arguments should match the `types` argument - single elements or tuples of
+# the same arity.
 class ExternalInputIterator(object):
-    def __init__(self, batch_size, shape, types, kinds):
+    def __init__(self, batch_size, shape, types, kinds, disallow_zeros = None):
         try:
             self.length = len(types)
         except TypeError:
             types = (types,)
             kinds = (kinds,)
             self.length = 1
+        if not disallow_zeros:
```

Comment:
Suggestion: how about a sanity check that `disallow_zeros` is `False` when generating `bool`s?